lib/routes/account.js

```diff
@@ -972,27 +1062,42 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified ||
-            Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
-          return reply({})
-        }
-        customs.check(
-          request.app.clientAddress,
-          sessionToken.email,
-          'recoveryEmailResendCode')
-          .then(
-            mailer.sendVerifyCode.bind(
-              mailer,
-              sessionToken,
-              sessionToken.emailCode,
-              {
-                service: service,
-                redirectTo: request.payload.redirectTo,
-                resume: request.payload.resume,
-                acceptLanguage: request.app.acceptLanguage
+
+        return db.sessionTokenWithVerificationStatus(sessionToken.id)
+          .then(function (token) {
+
+            // Choose which type of email to send
+            var doSigninConfirmation = requestHelper.shouldEnableSigninConfirmation(sessionToken.uid.toString('hex'), config)
+            var func
+            if (!token.emailVerified || !doSigninConfirmation) {
+
+              // Check for blackout period
+              if (Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
+                return reply({})
               }
-            )
-          )
+
+              func = mailer.sendVerifyCode
+            } else {
+              func = mailer.sendVerifyLoginEmail
+            }
+
+            return customs.check(
+              request.app.clientAddress,
+              sessionToken.email,
+              'recoveryEmailResendCode')
+              .then(func.bind(
+                mailer,
+                sessionToken,
+                doSigninConfirmation ? token.tokenVerificationId : sessionToken.emailCode, // Send tokenVerficationId, unless sign-in confirmation is disabled
```

Comment:
I can't think of any repercussions of this either. My original thinking was that in the new auth-server, if we always send a `tokenVerificationId`, the session and email get verified. While it is less code, it doesn't cover the scenario above. I can update this function.